packages/host/tests/integration/components/serialization-test.gts

```diff
@@ -3645,4 +3986,434 @@ module('Integration | serialization', function (hooks) {
+      assert.strictEqual(card.firstName, 'Burcu');
+      assert.deepEqual(card.friendPets, [], 'relationship is an empty array');
+    });
+
+    test('can deserialize a computed linksToMany relationship that does not include all the related resources', async function (assert) {
+      let {
+        field,
+        contains,
+        linksTo,
+        linksToMany,
+        Card,
+        createFromSerialized,
+        relationshipMeta,
+        serializeCard,
+      } = cardApi;
+      let { default: StringCard } = string;
+      class Pet extends Card {
+        @field name = contains(StringCard);
+      }
+      class Friend extends Card {
+        @field firstName = contains(StringCard);
+        @field pets = linksToMany(Pet);
+      }
+      class Person extends Card {
+        @field firstName = contains(StringCard);
+        @field friend = linksTo(Friend);
+        @field friendPets = linksToMany(Pet, {
+          computeVia: function (this: Person) {
+            return this.friend?.pets;
+          },
+        });
+      }
+      await shimModule(`${realmURL}test-cards`, { Pet, Friend, Person });
+      let doc: LooseSingleCardDocument = {
+        data: {
+          type: 'card',
+          attributes: { firstName: 'Burcu' },
+          relationships: {
+            friend: { links: { self: `${realmURL}Friend/hassan` } },
+            'friendPets.0': { links: { self: `${realmURL}Pet/mango` } },
+            'friendPets.1': { links: { self: `${realmURL}Pet/vanGogh` } },
+          },
+          meta: {
+            adoptsFrom: { module: `${realmURL}test-cards`, name: 'Person' },
+          },
+        },
+        included: [
+          {
+            id: `${realmURL}Pet/mango`,
+            type: 'card',
+            attributes: { name: 'Mango' },
+            meta: {
+              adoptsFrom: { module: `${realmURL}test-cards`, name: 'Pet' },
+            },
+          },
+        ],
+      };
+      let card = await createFromSerialized<typeof Person>(
+        doc.data,
+        doc,
+        undefined
+      );
+
+      try {
+        card.friend;
+        throw new Error(`expected error not thrown`);
+      } catch (err: any) {
+        assert.ok(err instanceof NotLoaded, 'NotLoaded error thrown');
+        assert.ok(
+          err.message.match(
+            /The field Person\.friend refers to the card instance https:\/\/test-realm\/Friend\/hassan which is not loaded/,
+            'NotLoaded error describes field not loaded'
+          )
+        );
+      }
+
+      try {
+        card.friendPets;
+        throw new Error(`expected error not thrown`);
+      } catch (err: any) {
+        assert.ok(err instanceof NotLoaded, 'NotLoaded error thrown');
+        assert.ok(
+          err.message.match(
+            /The field Person\.friendPets refers to the card instance https:\/\/test-realm\/Pet\/vanGogh which is not loaded/,
+            'NotLoaded error describes field not loaded'
+          )
+        );
```

Comment:
Should the error here refer to the friend card or the friendPet card that was not loaded?